inc/ustring.bi

```diff
@@ -0,0 +1,681 @@
+  #endif  
+  #ifndef zstr                                        'zero terminated ansi string (* bytes)
+    #define zstr zstring
+  #endif  
+#ENDIF
+
+
+
+'***********************************************************************************************
+' replacement for Jos� Roca�s CWstr, if WinFBX is not present or is not used.
+' this is mostly a clone of his code without Windows dependency and adapted to
+' work wih Linux as well.
+'***********************************************************************************************
+
+
+#include once "crt\mem.bi"
+
+
+#IFNDEF __FB_DOS__
+
+
+NAMESPACE JK                                          'obviously i cannot use "AFX" here
+TYPE DWSTR_ AS DWSTR                                  'Forward reference
+
+
+'***********************************************************************************************
+' DWSTR CLASS
+'***********************************************************************************************
+
+#if sizeof(wstring) = 2
+TYPE DWSTR extends wstring
+  Private:
+    m_Capacity AS Ulong                               'the total size of the buffer
+    m_GrowSize AS LONG = 260 * SizeOf(WSTRING)        'how much to grow the buffer by when required
+
+  Public:
+    m_pBuffer AS UBYTE PTR                            'pointer to the buffer
+    m_BufferLen AS ulong                              'length in bytes of the current string in the buffer
+    u_size      as long = SizeOf(WSTRING)
+
+    DECLARE CONSTRUCTOR
+    DECLARE CONSTRUCTOR (BYVAL nChars AS ulong, BYVAL nCodePage AS ulong)
+    DECLARE CONSTRUCTOR (BYVAL pwszStr AS WSTRING PTR)
+    DECLARE CONSTRUCTOR (BYREF ansiStr AS STRING, BYVAL nCodePage AS ulong = 0)
+    DECLARE CONSTRUCTOR (BYREF cws AS DWSTR)
+    DECLARE CONSTRUCTOR (BYVAL n AS LONGINT)
+    DECLARE CONSTRUCTOR (BYVAL n AS DOUBLE)
+
+    DECLARE DESTRUCTOR
+
+    DECLARE SUB ResizeBuffer (BYVAL nValue AS ulong)
+    DECLARE FUNCTION AppendBuffer (BYVAL addrMemory AS ANY PTR, BYVAL nNumBytes AS ulong) AS BOOLEAN
+    DECLARE FUNCTION InsertBuffer (BYVAL addrMemory AS ANY PTR, BYVAL nIndex AS ulong, BYVAL nNumBytes AS ulong) AS BOOLEAN
+
+    DECLARE PROPERTY Size () AS LONG
+    DECLARE PROPERTY Size (BYVAL nValue AS LONG)
+    DECLARE PROPERTY GrowSize () AS LONG
+    DECLARE PROPERTY GrowSize (BYVAL nValue AS LONG)
+    DECLARE PROPERTY Capacity () AS ulong
+    DECLARE PROPERTY Capacity (BYVAL nValue AS ulong)
+    DECLARE PROPERTY SizeAlloc (BYVAL nChars AS ulong)
+    DECLARE PROPERTY SizeOf () AS ulong
+    DECLARE PROPERTY CodePage () AS ulong
+    DECLARE PROPERTY CodePage (BYVAL nCodePage AS ulong)
+
+    DECLARE SUB Clear
+
+    DECLARE SUB Add (BYREF cws AS DWSTR)
+    DECLARE SUB Add (BYVAL pwszStr AS WSTRING PTR)
+    DECLARE SUB Add (BYREF ansiStr AS STRING, BYVAL nCodePage AS ulong = 0)
+
+    DECLARE OPERATOR [] (BYVAL nIndex AS ulong) byref AS USHORT
+
+    DECLARE OPERATOR CAST () BYREF AS WSTRING
+    DECLARE OPERATOR CAST () AS ANY PTR
```

Comment:
Doesn't this allow the `ustring` to be passed to any pointer type and effectively disables type checking?